hack/generator/pkg/codegen/storage/property_converter.go

```diff
@@ -90,6 +91,11 @@ func (p *PropertyConverter) useBaseTypeForEnumerations(
 func (p *PropertyConverter) shortCircuitNamesOfSimpleTypes(
 	tv *astmodel.TypeVisitor, tn astmodel.TypeName, ctx interface{}) (astmodel.Type, error) {
 
+	// for nonlocal packages, preserve the name as is
+	if _, ok := tn.PackageReference.AsLocalPackage(); !ok {
```

Comment:
We don't generate the storage types in the Golden files tests right now. Once we do, this issue would be caught there by the existing golden file test that has a variety of `v1.JSON` types.

Given that, I didn't try to add a test specifically for this. Once we enable storage type generation in golden files we should be covered.

Reply:
I think we should be able to add a test - the test coverage of the conversions package is intended to catch issues here, rather than forcing a more involved analysis if/when it fails as a part of storage conversion.

Reply:
Have added a test - which found some interesting edge cases we needed to handle.